client/src/app/project-registration/project-registration-form/project-registration-form.component.ts

```diff
@@ -62,6 +68,9 @@ export class ProjectRegistrationFormComponent implements OnInit {
 
     if (queryParam.has(Identifier.DOI)) {
```

Comment:
The `ngOnInit` function is very long and has too much detail.
 This bit with `projectFormData` can be extracted to a function, as can the  bit after it.